pkg/roll/run_update.go

```diff
@@ -684,7 +713,31 @@ func (u *update) countHealthy(id rcf.ID, checks map[types.NodeName]health.Result
 		ret.Unknown = ret.Desired - ret.Current
 	}
 
+	var eligibleNodes []types.NodeName
+	if rcFields.AllocationStrategy == rcf.DynamicStrategy {
+		fmt.Println("calling bro")
+		eligibleNodes, err = u.scheduler.EligibleNodes(rcFields.Manifest, rcFields.NodeSelector)
+		if err != nil {
+			return ret, util.Errorf("could not enumerate eligible nodes for %s: %s", rcFields.ID, err)
+		}
+	}
+
 	for _, pod := range currentPods {
+		if rcFields.AllocationStrategy == rcf.DynamicStrategy {
+			// don't count this pod if it's ineligible on a dynamic RC, it will be undergoing a node transfer soon and therefore might be unscheduled
+			// beneath the rolling update
+			nodeEligible := false
+			for _, eligibleNode := range eligibleNodes {
+				if eligibleNode == pod.Node {
+					nodeEligible = true
+					break
+				}
+			}
+			if !nodeEligible {
+				continue
+			}
```

Comment:
I can't tell what this if block does. Doesn't the code go the next iteration of the loop anyway?

Reply:
i'll add a comment to clarify. Basically I don't want to increment `ret.Healthy` or `ret.Real` if the node isn't eligible

Reply:
Nevermind. I totally misinterpreted the truncated file on github and thought the loop ended 🙃